flink-ai-flow/examples/workflow_on_event/README.md

```diff
@@ -0,0 +1,158 @@
+
+Whenever hourly data is produced, it sends an event with the following format:
+```json
+{
+  "namespace": "workflow_on_event",
+  "event_type": "DATA_EVENT",
+  "key": "hourly_data",
+  "value": "ready",
+  "context": "2021-09-13T15"
+}
+``` 
+The `context` is the hour of the produced data in iso format.
+
+### Hourly Preprocess Workflow
+
+The Hourly Preprocess Workflow runs whenever the following event arrives:
+ ```json
+{
+  "namespace": "workflow_on_event",
+  "event_type": "DATA_EVENT",
+  "key": "hourly_data",
+  "value": "ready",
+  "sender": "*"
+}
+``` 
+The context of the WorkflowExecution is the `context` field of the event. So the Hourly Preprocess Workflow
+should be triggered when the Hourly Data Workflow produced an hourly dataset.
+
+The Hourly Preprocess Workflow will transform the hourly data and copy it to the uri of the `daily_data` dataset meta.
+
+At the end of the workflow, it checks if all the hourly datasets are ready for that day. If so, it sends an event with
+the following format:
+```json
+{
+  "namespace": "workflow_on_event",
+  "event_type": "DATA_EVENT",
+  "key": "daily_data",
+  "value": "ready",
+  "context": "2021-09-13"
+}
+``` 
+The `context` is the day when all the hourly data is ready.
+
+### Daily Workflow
+
+The Daily Workflow is triggered by the following event:
+ ```json
+{
+  "namespace": "workflow_on_event",
+  "event_type": "DATA_EVENT",
+  "key": "daily_data",
+  "value": "ready",
+  "sender": "*"
+}
+``` 
+The context of the WorkflowExecution is the `context` field of the event. 
+
+The Daily Workflow train a new model with the data from the day specified in the event context. It loads a base
+model for training if there is a deployed/validated/generated model train today or yesterday. Therefore, a new model
+is registered every day with the model name `daily_model_{date}`. You can train multiple model versions of the model if
+you run the Daily Workflow for the same day multiple times.
+
+After training, we have a validate job to validate the model with the validating dataset. It compares the score with the
+current model version with the previous model version and mark the current model version as validated if the score of 
+The current model version is higher.
+
+## How to Run
+
+In this section, we show step-by-step instructions to start the workflows in the project. We strongly suggest that you
+have successfully followed the [Quick Start](https://github.com/alibaba/flink-ai-extended/wiki/Quick-Start) so that you 
+are comfortable with starting an AIFlow server and running a workflow.
+
+All the commands in this section should run at the `flink-ai-flow/examples/workflow_on_event` directory. 
+
+### Initialization
+
+We start by initializing the dataset metadata and model metadata. To do that, you can simply run
+
+```bash
+python workflows/init/init.py
+```
+ 
+If the command runs successfully, you should see the registered dataset and model on the AIFlow web UI or database. 
+
```

Comment:
It would be better if we can add why we choose the following order to start these 3 workflows.

Reply:
Explanation is added in 8d41949